packages/model-viewer/src/features/scene-graph/texture.ts

```diff
@@ -15,42 +15,36 @@
 
 import {Texture as ThreeTexture} from 'three';
 
-import {GLTFElement} from '../../three-components/gltf-instance/gltf-2.0.js';
+import {Texture as GLTFTexture} from '../../three-components/gltf-instance/gltf-2.0.js';
 
 import {Texture as TextureInterface} from './api.js';
 import {Image} from './image.js';
 import {Sampler} from './sampler.js';
-import {$gltfTexture, $threeTexture, TextureInfo} from './texture-info.js';
-import {$correlatedObjects, $onUpdate, $sourceObject, ThreeDOMElement} from './three-dom-element.js';
+import {$sourceObject, ThreeDOMElement} from './three-dom-element.js';
 
 
 
-const $source = Symbol('source');
+const $image = Symbol('image');
 const $sampler = Symbol('sampler');
 
 /**
  * Material facade implementation for Three.js materials
  */
 export class Texture extends ThreeDOMElement implements TextureInterface {
-  private[$source]: Image;
+  private[$image]: Image;
   private[$sampler]: Sampler;
 
-  constructor(textureInfo: TextureInfo) {
+  constructor(
+      onUpdate: () => void,
+      sampler: Sampler,
+      image: Image,
+      texture: ThreeTexture|null,
+      gltfTexture: GLTFTexture|null,
+  ) {
     super(
-        textureInfo.onUpdate,
-        textureInfo[$gltfTexture],
-        new Set<ThreeTexture>([textureInfo[$threeTexture]!]));
-    this[$sampler] = new Sampler(textureInfo);
-    this[$source] = new Image(textureInfo);
-  }
-
-  applyNewTextureInfo(textureInfo: TextureInfo): void {
-    (this[$onUpdate] as () => void) = textureInfo.onUpdate;
-    (this[$sourceObject] as GLTFElement) = textureInfo[$gltfTexture];
-    (this[$correlatedObjects] as Set<ThreeTexture>) =
-        new Set<ThreeTexture>([textureInfo[$threeTexture]!]);
-    this[$sampler] = new Sampler(textureInfo);
-    this[$source] = new Image(textureInfo);
+        onUpdate, gltfTexture, new Set<ThreeTexture>(texture ? [texture] : []));
```

Comment:
fixes a source object problem, glTF TextureInfo was being supplied as the source data but should have been glTF Texture